blog/2020-09-25-gh-release.md

```diff
@@ -0,0 +1,165 @@
+The `release` event contains the `upload_url` so we can simply pass it into the `@actions/upload-release-asset` action.
+
+## Local actions for code-reuse
+
+Whilst having modular workflows was a good experience, it did create some code duplication between the different workflows.
+With GitHub actions, the unit of code reuse is a GitHub action.
+However, releasing a GitHub action to the marketplace felt like an overkill for this problem.
+Fortunately, a step can [reference](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#example-using-action-in-the-same-repository-as-the-workflow) an action that is stored within the same repository.
+
+This allowed me to extract [two](https://github.com/comit-network/comit-rs/tree/dev/.github/actions) otherwise duplicated sections of different workflows into dedicated actions and reuse them.
+
+## Releasing several components from the same repository
+
+The `comit-rs` repository is a Cargo workspace and therefore home to several crates.
+Two of them are binaries that are released using the workflows discussed in this post.
+In the future, we might also release crates to https://crates.io/ from the repository.
+
+Regarding the two binaries, I had to make a decision in the workflow design:
+
+The release process is very similar, do I make one configurable workflow or duplicate most of the code?
+
+In the end, I settled with two separate workflows that are almost identical.
+While one could complain about this duplicated code, I don't think it is very harmful in this case.
+
+1. The two workflows are similar for now but that doesn't mean they always will be.
+There is already one difference: `cnd` is also published as a docker image, whereas `nectar` isn't.
+It is little details like that which would require configuration of the workflow, making it harder to maintain.
+2. Testing workflows is mostly a manual process.
+Not having to worry about different configuration combinations greatly simplifies maintenance.
+3. It is much easier to follow along what the workflow does if there are several "hardcoded" values like [the prefix of the tag](https://github.com/comit-network/comit-rs/blob/acabab36058b4e3d3ae54bde10fd7ce5db66f1a5/.github/workflows/release-cnd.yml#L32-L37).
+
+## Testing workflows
+
+Testing workflows can be cumbersome because all we are writing is untyped yaml.
+I ended up creating a fork the repository in my personal account for the testing.
+This allowed me to iterate fairly quickly because I was able to remove all branch protections from the repository.
+As such, I could force push to `dev` and `master` and test my workflows using the real trigger events.
+This is something I could have never done in the main repository and therefore it greatly sped up development.
+
+## Conclusion
+
+This turned out a bit longer than I originally expected!
```

Comment:
I feel the conclusions as it is does not add much value, maybe actually sum up the post in 2-3 sentences?
(the last sentence is fine, but on its own it does not add much value)

Reply:
I didn't want to make it any longer by summarizing stuff.